core/src/main/scala/org/apache/spark/storage/ShuffleBlockFetcherIterator.scala

```diff
@@ -444,36 +445,36 @@ final class ShuffleBlockFetcherIterator(
               throwFetchFailedException(blockId, address, e)
           }
 
-          input = streamWrapper(blockId, in)
-          // Only copy the stream if it's wrapped by compression or encryption, also the size of
-          // block is small (the decompressed block is smaller than maxBytesInFlight)
-          if (detectCorrupt && !input.eq(in) && size < maxBytesInFlight / 3) {
-            val originalInput = input
-            val out = new ChunkedByteBufferOutputStream(64 * 1024, ByteBuffer.allocate)
-            try {
+          try {
+            input = streamWrapper(blockId, in)
```

Comment:
You do want to close `in` if this call throws an exception, though.

Reply:
in only need to be closed when we actually copy streams meaning that only when `(detectCorrupt && !input.eq(in) && size < maxBytesInFlight / 3)` is true.